corehq/motech/fhir/tasks.py

```diff
@@ -394,5 +398,252 @@ def create_parent_indices(
+
+    # get fhir ids for appointments currently in commcare
+    appointment_fhir_ids = [
+        appointment_case.get_case_property('fhir_id') for appointment_case in appointment_cases
+    ]
+
+    for patient in patient_cases:
+        patient_helper = CaseHelper(case=patient, domain=domain)
+        patient_fhir_id = patient.get_case_property('patient_fhir_id')
+        if not patient_fhir_id:
+            given = patient.get_case_property('given_name')
+            family = patient.get_case_property('family_name')
+            birthdate = patient.get_case_property('birthdate')
+            patient_fhir_id = get_patient_fhir_id(given, family, birthdate, access_token)
+
+            if patient_fhir_id is not None:
+                patient_helper.update({'properties': {
+                    'patient_fhir_id': patient_fhir_id,
+                }})
+
+        epic_appointments_to_add = []
+        epic_appointments_to_update = []
+        # Get all appointments for patient from epic
+        epic_appointment_records = get_epic_appointments_for_patient(patient_fhir_id, access_token)
+        for appointment in epic_appointment_records:
+            appointment_resource = appointment.get('resource')
+            appointment_id = appointment_resource.get('id')
+            if appointment_id and appointment_id not in appointment_fhir_ids:
+                epic_appointments_to_add.append(appointment)
+            elif appointment_id:
+                epic_appointments_to_update.append(appointment)
+
+        # Add new appointments to commcare
+        for appointment in epic_appointments_to_add:
+            appointment_create_helper = CaseHelper(domain=domain)
+            appointment_resource = appointment.get('resource')
+            if appointment_resource is not None:
+                appointment_description = appointment_resource.get('description') or 'NO DESCRIPTION LISTED'
+                appointment_fhir_timestamp = appointment_resource.get('start')
+                appointment_date, appointment_time = convert_utc_timestamp_to_date_and_time(
+                    appointment_fhir_timestamp)
+                appointment_fhir_id = appointment_resource.get('id')
+                reason = None
+                practitioner = None
+                for p in appointment_resource.get('participant'):
+                    actor = p.get('actor')
+                    if actor and 'Practitioner' in actor.get('reference'):
+                        practitioner = actor.get('display')
+                        break
+                reason_code = appointment_resource.get('reasonCode')
+                if reason_code and reason_code[0] is not None:
+                    reason = reason_code[0].get('text')
+            host_case_id = patient.get_case_property('case_id')
+            appointment_case_data = {
+                'case_name': appointment_fhir_timestamp + appointment_description,
```

Comment:
int: add name formatting
```suggestion
                'case_name': f"[{appointment_fhir_timestamp}]: {appointment_description}",
```